src/pages/[marketplace]/deals/new.tsx

```diff
@@ -0,0 +1,97 @@
+import DealForm, { DealFormInput } from "@components/DealForm";
+import Layout from "@components/Layout";
+import { Link } from "@components/Link";
+import { useCredixClient } from "@credix/credix-client";
+import { PublicKey } from "@solana/web3.js";
+import { numberFormatter, toProgramAmount } from "@utils/format.utils";
+import Big from "big.js";
+import message from "message";
+import { useRouter } from "next/router";
+import { NextPageWithLayout } from "pages/_app";
+import React, { ReactElement, useEffect } from "react";
+import { useStore } from "state/useStore";
+
+const New: NextPageWithLayout = () => {
+	const router = useRouter();
+	const { marketplace } = router.query;
+	const client = useCredixClient();
+	const fetchMarket = useStore((state) => state.fetchMarket);
+	const market = useStore((state) => state.market);
+
+	useEffect(() => {
+		fetchMarket(client, marketplace as string);
+	}, [client, fetchMarket, marketplace]);
+
+	const onSubmit = async ({
+		principal,
+		financingFee,
+		timeToMaturity,
+		borrower,
+		dealName,
+	}: DealFormInput) => {
+		const formattedPrincipal = numberFormatter.format(principal);
+		const hide = message.loading({ content: `Creating deal for ${formattedPrincipal} USDC` });
+		const borrowerPK = new PublicKey(borrower);
```

Comment:
Maybe some validation on the `borrower` input? If it's not a valid PublicKey this will throw an error.